docs/agent/config/v3.mdx

```diff
@@ -0,0 +1,571 @@
+---
+title: "Version 3"
+---
+
+import GlobalConfiguration from "./_global-configuration.mdx";
+
+## Full Example
```

Comment:
This "Full example" section is the main set of changes, where I:

- removed references to sections that no longer make sense under `endpoints`
- updated the examples to use the new `endpoints` formats, with relevant updated examples for each new case
- fixed links now that we have v2/v3
- fixed copy-text to mark some items as "deprecated", and with explanatory text.